### DISCUSSION TITLE
Return traceid (and, in general, context info) into header response

### DISCUSSION THREAD

**Discussion by ziomill** [open-telemetry/opentelemetry-java#4211] [Q&A] [answered]
Hello,

We are using java automatic instrumentation in some Java SpringBoot microservices. We have also adopted the "**W3C Trace Context**" strategy for Context propagation.

We would return some context info (like the **traceid**) into the Response Headers. Is there a way to access to these params, in a standard way, through the OpenTelemetry SDK? 

Of course, we could simply read the headers by their key name (ex.: traceparent) but, in my opinion, this will introduce a strong coupling with the "**W3C Trace context**" strategy, which we would avoid if possible.

Thank you

**Comment by jkwatson** [comment, marked as answer]
`Span.current().getSpanContext()` should give you what you need, I think.

**Comment by ziomill** [comment]
Hi @jkwatson, 
It's exactly what I was looking for:

```SpanContext spanContext = Span.current().getSpanContext();
   String traceId = spanContext.getTraceId();
   String parentId = spanContext.getSpanId();
   String traceFlags = spanContext.getTraceFlags().asHex();
```

Thank you.
